File: app/src/main/java/com/glitch/detectivesearch/cases/CasesAdapter.java
package com.glitch.detectivesearch.cases;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.glitch.detectivesearch.ListItemClickListener;
import com.glitch.detectivesearch.R;

public class CasesAdapter extends RecyclerView.Adapter<CasesViewHolder> {
    private int mNumberItems;
    private CasesInfo[] items;
    private ListItemClickListener mOnClickListener;
    int ID = 0;

    public CasesAdapter(int numberOfItems, CasesInfo[] itemList, ListItemClickListener listener) {
        mNumberItems        = numberOfItems;
        items               = itemList;
        mOnClickListener    = listener;
    }

    @Override
    public CasesViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        ID= R.layout.list_view;

        int            layoutIdForListItem             = ID;
        LayoutInflater inflater                        = LayoutInflater.from(context);
        boolean        shouldAttachToParentImmediately = false;

        View view = inflater.inflate(layoutIdForListItem, parent, shouldAttachToParentImmediately);
        CasesViewHolder viewHolder = new CasesViewHolder(view, mOnClickListener);

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(CasesViewHolder holder, int position) {
        holder.bind(items[position]);
    }

    @Override
    public int getItemCount() {
        return items.length;
    }

    public void setData(CasesInfo[] data) {
        items        = data;
        mNumberItems = data.length;
        notifyDataSetChanged();
    }
    public CasesInfo[] getdata() {
        return items;
    }
}

File: app/src/main/java/com/glitch/detectivesearch/ui/cases/CasesAdapter.kt
package com.glitch.detectivesearch.ui.cases

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.glitch.detectivesearch.R
import com.glitch.detectivesearch.data.model.response.Case
import com.glitch.detectivesearch.databinding.ItemCaseBinding

class CasesAdapter(
	private val onCaseClick: (Int) -> Unit
) : RecyclerView.Adapter<CasesAdapter.CasesViewHolder>() {

	private val caseList = mutableListOf<Case>()

	override fun onCreateViewHolder(
		parent: ViewGroup, viewType: Int
	): CasesAdapter.CasesViewHolder {
		val binding = ItemCaseBinding.inflate(LayoutInflater.from(parent.context), parent, false)
		return CasesViewHolder(binding, onCaseClick)
	}

	override fun onBindViewHolder(holder: CasesAdapter.CasesViewHolder, position: Int) {
		holder.bind(caseList[position])
	}

	override fun getItemCount(): Int {
		return caseList.size
	}

	class CasesViewHolder(
		private val binding: ItemCaseBinding, val onCaseClick: (Int) -> Unit
	) : RecyclerView.ViewHolder(binding.root) {
		fun bind(case: Case) {
			with(binding) {
				if (case.isCaseEnabled == "false") {
					ivFile.setImageResource(R.drawable.file_false)
				}
				if (case.isCaseEnabled == "true") {
					ivFile.setImageResource(R.drawable.file)
				}
				if (case.isCaseEnabled == "done") {
					ivFile.setImageResource(R.drawable.file_done)
				}
				tvProductName.text = case.caseName

				root.setOnClickListener {
					onCaseClick(case.id)
				}
			}
		}
	}
	fun updateList(list: List<Case>) {
		caseList.clear()
		caseList.addAll(list)
		notifyItemRangeChanged(0, list.size)
	}
}